modules/javafx.controls/src/main/java/javafx/scene/control/SkinBase.java

```diff
@@ -238,7 +241,79 @@ protected final void registerChangeListener(ObservableValue<?> property, Consume
         return lambdaChangeListenerHandler.unregisterChangeListeners(property);
     }
 
+    /**
+     * Subclasses can invoke this method to register that they want to listen to
+     * invalidation events for the given observable. Registered {@link Consumer} instances
+     * will be executed in the order in which they are registered.
+     *
+     * @param observable the observable to observe for invalidation events
+     * @param consumer the consumer
```

Comment:
Could be just me because I'm unfamiliar with this API, but I'd like to see an explanation for what the consumer is used for. If the consumer is executed on invalidation events on this observable, then clearly state it. Maybe the doc could be something like:

    Registers an action to take when an {@code Observable} is invalidated. 
    An {@code InvalidationListener} is registered on the given {@code observable} and the {@code consumer} is run 
    whenever the listener sends an invalidation event. If multiple consumers are registered on the same observable, 
    they are run in the order in which they were registered.

    @param observable the observable to observe for invalidation events
    @param consumer the action to take when the observable is invalidated

Since it's a `protected final` method, I don't think that the "Subclasses can use this..." portion is necessary - that's the only use of these methods.

Reply:
The changes look good.

We use code tags when we refer to terms found in code, such as class, method and argument names, and literal like `true`, `class` and `null`. The docs have not been consistent, I know, but wherever we touch we try to fix it along the way. It's just not worth to do fixes specifically for that.